drivers/serial/uart_gecko.c

```diff
@@ -354,19 +449,81 @@ static const struct uart_driver_api uart_gecko_driver_api = {
 			gpioModePushPull, 1				       \
 	}
 
-#define GECKO_UART_PINS(idx)						       \
+#define GECKO_UART_RX_TX_PINS(idx)					       \
 	.pin_rx = PIN_UART_RXD(idx),					       \
 	.pin_tx = PIN_UART_TXD(idx),
 
+#ifdef UART_GECKO_HW_FLOW_CONTROL
+
+#ifdef CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION
+#define GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)				       \
+	.loc_rts = COND_CODE_1(DT_INST_PROP(idx, hw_flow_control),	       \
+		   (DT_INST_PROP_BY_IDX(idx, location_rts, 0)),		       \
+		   (0)),						       \
+	.loc_cts = COND_CODE_1(DT_INST_PROP(idx, hw_flow_control),	       \
+		   (DT_INST_PROP_BY_IDX(idx, location_cts, 0)),		       \
+		   (0)),
+#define VALIDATE_GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)			       \
+	COND_CODE_1(DT_INST_PROP(idx, hw_flow_control),			       \
+		    (BUILD_ASSERT(DT_INST_NODE_HAS_PROP(idx, location_rts) &&  \
+				  DT_INST_NODE_HAS_PROP(idx, location_cts),    \
+			"DTS location-rts and location-cts are mandatory")),   \
+		    ())
+#else /* CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION */
+/* Hardware flow control not supported for these SOCs */
+#define GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)
+#define VALIDATE_GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)
+#endif /* CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION */
+
+#define PIN_UART_RTS(idx)						       \
+	COND_CODE_1(DT_INST_PROP(idx, hw_flow_control),			       \
+	({								       \
+		DT_INST_PROP_BY_IDX(idx, location_rts, 1),		       \
+			DT_INST_PROP_BY_IDX(idx, location_rts, 2),	       \
+			gpioModePushPull, 1				       \
+	}),								       \
+	({}))
```

Comment:
`{}` is a GNU extension. Since we're planning to support other toolchains it would be better to change it to the standard `{0}`.

Reply:
done